src/reporter.cc

```diff
@@ -157,6 +158,175 @@ Napi::Value send_span(const Napi::CallbackInfo& info, send_generic_span_t send_f
 
 }
 
+//
+// sendMetric(name, object)
+//
+// only the first argument is required for an increment call.
+//
+// name - the name of the metric
+// object - an object containing optional parameters
+// object.count - the number of observations being reported (default: 1)
+// object.addHostTag - boolean - {host: hostname} to tags.
+// object.tags - an object containing {tag: value} pairs.
+// object.value - if present this call is a valued-based call and this contains
+//                the value, or sum of values if count is greater than 1, being
+//                reported.
+//
+// there are two types of metrics:
+//   1) count-based - the number of times something has occurred (no value associated with this metric)
+//   2) value-based - a specific value is being reported (or a sum of values)
+//
+//
+// simplest forms:
+// sendMetric('my.little.count')
+// sendMetric('my.little.value', {value: 234.7})
+//
+// to report two observations:
+// sendMetric('my.little.count', {count: 2})
+// sendMetric('my.little.value', {count: 2, value: 469.4})
+//
+// to supply tags that can be used for filtering:
+// sendMetric('my.little.count', {tags: {status: error}})
+//
+// to add a host name tag:
+// sendMetric('my.little.count', {addHostTag: true, tags: {status: error}})
+//
+Napi::Value sendMetric (const Napi::CallbackInfo& info) {
+  Napi::Env env = info.Env();
+
+  // check args
+  if (info.Length() == 0) {
+    Napi::TypeError::New(env, "sendMetric() requires 1 or 2 arguments")
+        .ThrowAsJavaScriptException();
+    return env.Null();
+  } else if (!info[0].IsString()) {
+      Napi::TypeError::New(env, "sendMetric(name) name argument must be a string")
+          .ThrowAsJavaScriptException();
+      return env.Null();
+  }
+
+  // default the options object if not supplied
+  Napi::Object o;
+  if (info.Length() == 1) {
+    o = Napi::Object::New(env);
+  } else if (info[1].IsObject() && !info[1].IsArray()) {
+    o = info[1].ToObject();
+  } else {
+    Napi::TypeError::New(env, "sendMetric(name, params) params must be a plain object")
+        .ThrowAsJavaScriptException();
+    return env.Null();
+  }
+
+  // name
+  std::string name(info[0].As<Napi::String>());
+
+  // count
+  int count = 1;
+  Napi::Value v = o.Get("count");
+  if (v.IsNumber()) {
+    count = v.As<Napi::Number>().Int64Value();
+  }
+
+  // value
+  // the presence of "value" indicates that this is a summary metric (values)
+  // and not an increment metric (count of occurrences).
+  bool is_summary = false;
+  double value = 0;
+  if (o.Has("value")) {
+    v = o.Get("value");
+    if (!v.IsNumber()) {
+      Napi::TypeError::New(env, "sendMetric options.value must be a number")
+          .ThrowAsJavaScriptException();
+      return env.Null();
+    }
+    value = v.As<Napi::Number>().DoubleValue();
+    is_summary = true;
+  }
+
+
+  // host_tag
+  bool host_tag = false;
+  if (o.Has("addHostTag")) {
+    host_tag = o.Get("addHostTag").ToBoolean().Value();
```

Comment:
ToBoolean() coerces the value to a boolean and everything javascript can be interpreted as a boolean. Often functions refer to passing a "truthy" or a "falsey" value because it's rare that a true boolean is needed.